Checking protonationstate of GLU, ASP and LYS

dafwb opened the pull request:
Hello,

I had an issue with a protonated glutamic acid and added some lines, to also check the protonation state for these residues.
Sorry for the doubled request, the first one was not correctly copied to the IDE.

Best,
Daniel

dafwb committed:
Add check of protonationstates for GLU, ASP, LYS

For the residues GLU, ASP, and LYS the protonation state is also checked and residues are renamed, if that is necessary.

dafwb committed:
update on assign_protonationstates

Shifted if-statetments to the right level

hainm commented:
Dear @dafwb, this repo is outdated compared to version in AmberTools.
But thanks for your PR.

I ping @dacase to have a look at your PR.

dafwb commented:
For my local installation I also added a function to only change residues of a given list for constant pH simulations:

```
    def constph_list(self, cpH_list_str):
        """ Update AS4, GL4, HIP for constph.
    
        Returns
        -------
        parm : updated `parm`
        """
        if "," in cpH_list_str:
            cpH_list = [int(i)-1 for i in cpH_list_str.split(",")]
        elif "+" in cpH_list_str:
            cpH_list = [int(i)-1 for i in cpH_list_str.split("+")]
        else:
            #assumin list only contains one value
            cpH_list = int(cpH_list_str)
        cpH_list.sort()
        for residue in self.parm.residues:
            if (residue.name == 'ASP' or residue.name == 'ASH') and residue.idx in cpH_list:
                residue.name = 'AS4'
            elif (residue.name == 'GLU' or residue.name == 'GLH') and residue.idx in cpH_list:
                residue.name = 'GL4'
            elif residue.name[:2] == 'HI' and residue.idx in cpH_list:
                residue.name = 'HIP'
            elif residue.idx > cpH_list[-1]:
                break #That saves time for systems, that are already solvated.
            else:
                pass
        self.parm.strip(':AS4@HD2')
        self.parm.strip(':GL4@HE2')
        return self
```
not sure, if that would also be helpful for others.

Commit:
Updated pdb4amber.py for protonationstates of ARG,GLU and LYS, and option of handing over a list of residues for constant pH simt

hainm commented:
Thanks @dafwb. I am still waiting for the feedback from @dacase
